pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -390,30 +389,25 @@ func (e *aksOperatorController) deployAKSOperator() error {
 func (e *aksOperatorController) generateSATokenWithPublicAPI(cluster *mgmtv3.Cluster) (string, bool, error) {
 	restConfig, err := e.getKubeConfig(cluster)
 
-	netDialer := net.Dialer{
-		Timeout:   30 * time.Second,
-		KeepAlive: 30 * time.Second,
-	}
-	serviceToken, err := generateSAToken(restConfig, netDialer.DialContext)
+	// TODO: there will be a merge conflict here with another PR
+	restConfig.Dial = (&net.Dialer{Timeout: 30 * time.Second, KeepAlive: 30 * time.Second}).DialContext
+	serviceToken, err := clusteroperator.GenerateSAToken(restConfig)
 	if err != nil {
 		var dnsError *net.DNSError
 		if stderrors.As(err, &dnsError) && !dnsError.IsTemporary {
 			return "", true, nil
 		}
-	}
-
-	return serviceToken, false, err
-}
 
-func generateSAToken(restConfig *rest.Config, dialer typesDialer.Dialer) (string, error) {
-	restConfig.Dial = dialer
+		// In the existence of a proxy, it may be the case that the following error occurs,
+		// in which case rancher should use the tunnel connection to communicate with the cluster.
+		var urlError *url.Error
+		if stderrors.As(err, &urlError) && urlError.Timeout() {
+			return "", true, nil
+		}
 
-	clientSet, err := kubernetes.NewForConfig(restConfig)
-	if err != nil {
-		return "", fmt.Errorf("error creating clientset: %v", err)
 	}
 
-	return util.GenerateServiceAccountToken(clientSet)
+	return serviceToken, false, err
 }
 
 func (e *aksOperatorController) getKubeConfig(cluster *mgmtv3.Cluster) (*rest.Config, error) {
```

Comment:
Not mission critical but maybe this could be renamed to `getRestConfig` for consistency with the other handlers

Reply:
Done.